src/components/Card.js

```diff
@@ -20,15 +30,32 @@ class Card {
     });
   }
 
-  _deleteCard = () => {
+  getId() {
+    return this._id;
+  }
+
+  handleDeleteClick = () => {
+    this._handleCardDelete(this);
+  };
+
+  deleteCard = () => {
     this._element.remove();
     this._element = null; // Helps with memory management.
   };
 
-  _likeCard = () => {
+  _likeCard = (evt) => {
     this._likeButton.classList.toggle("locations__card-like_active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.